src/com/facebook/buck/apple/AppleDescriptions.java

```diff
@@ -207,7 +211,18 @@ public static void populateCxxConstructorArg(
                     arg))
             .build();
 
-    output.srcs = arg.srcs;
+    if (arg.srcs.isPresent()) {
```

Comment:
I think that what you have is already reasonable.  Either way, you're effectively splitting based on extension.  The only difference is where.

Architecturally, we want to keep the parser as simple as possible and avoid putting rule-specific logic in there.